src/backend/src/contacts.rs

```diff
@@ -1,11 +1,51 @@
-use shared::types::contact::Contact;
+use ic_cdk::api::time;
+use shared::types::contact::{Contact, ContactError, CreateContactRequest, StoredContacts};
 
-use crate::{read_state, types::StoredPrincipal};
+use crate::{
+    mutate_state, read_state,
+    types::{Candid, StoredPrincipal},
+};
 
-// The contact services provide the
-// Make sure to stick to the following rules when implementing a contact service
-// 1. Do not expose the stored model StoredContacts
-// 2. Return the
+pub fn create_contact(request: CreateContactRequest) -> Result<Contact, ContactError> {
+    if request.name.trim().is_empty() {
+        return Err(ContactError::InvalidContactData);
+    }
+
+    let stored_principal = StoredPrincipal(ic_cdk::caller());
+    let current_time = time();
+
+    mutate_state(|s| {
+        // Get or create the user's contacts storage
+        let mut stored_contacts = match s.contact.get(&stored_principal) {
+            Some(candid_stored_contacts) => candid_stored_contacts.clone(),
+            None => StoredContacts {
+                contacts: Vec::new(),
+                update_timestamp_ns: current_time,
+            },
+        };
+
+        // Generate a new unique ID for the contact
+        // Using the current timestamp as ID ensures uniqueness
```

Comment:
Not it doesn't ensure uniqueness.  It looks as if the id really is generated from time.  That must be changed.